sceptre/helpers.py

```diff
@@ -218,10 +218,12 @@ def get_subclasses(class_type, directory=None):
     :returns: A dict of classes found.
     :rtype: dict
     """
-    if not isinstance(directory, str):
-        raise TypeError("'directory' object should be a str")
+    try:
+        glob_expression = os.path.join(directory, "*.py")
+    except (AttributeError, TypeError):
```

Comment:
Python 2 and 3 throw different exceptions when `directory` is not a string, so I covered both.

```sh
# Python 2
>>> os.path.join(1, '')
AttributeError: 'int' object has no attribute 'endswith'

# Python 3.5.2
>>> os.path.join(1, '')
TypeError: join() argument must be str or bytes, not 'int'
```